src/sql/azdata.proposed.d.ts

```diff
@@ -1790,6 +1795,77 @@ declare module 'azdata' {
 		 * The path of the parent node.
 		 */
 		parentNodePath: string;
+		/**
+		 * flag to indicate if the node supports filtering
+		 */
+		isFilteringSupported?: boolean;
+		/**
+		 * list of filters that are supported by the node
+		 */
+		defaultFilters?: NodeInfoFilterDefinition[];
+		/**
+		 * list of properties to filter children nodes on
+		 */
+		filters?: NodeInfoFilter[];
+	}
+
+	export interface NodeInfoFilterDefinition {
+		/**
+		 * The name of the filter property
+		 */
+		name: string;
+		/**
+		 * The type of the filter property
+		 */
+		type: NodeInfoFilterDataType;
+		/**
+		 * The list of options for the filter property if the type is enum
```

Comment:
Might be easier to use if you define a separate `NodeFilterEnumDefinition` (updating the name per my other comment below if needed). That way you could have options be a required property for that.